src/fmllauncher/java/net/minecraftforge/fml/loading/moddiscovery/ModFileInfo.java

```diff
@@ -54,6 +55,10 @@
     private final List<IModInfo> mods;
     private final Map<String,Object> properties;
     private final String license;
+    // Caches the manifest of the mod jar as parsing the manifest can be expensive for
+    // signed jars. The manifest needs to be an Optional<Manifest>, so a nullable Optional
+    // is probably the best option.
+    private volatile Optional<Manifest> manifest = null;
```

Comment:
If it doesn't matter that two threads hold different instances of the manifest (with the same content), this does not need to be volatile

Reply:
Yes it does. As soon as more than one thread can access a variable you _must_ use volatile (or synchronization). If you just use a normal field, it would be perfectly okay for other threads than the writing thread to _never_ see this new value.